R/remote.R

```diff
@@ -0,0 +1,58 @@
+
+isChildProcess <- function() {
+  !is.na(Sys.getenv("RSTUDIOAPI_IPC_REQUESTS_FILE", unset = NA))
+}
+
+callRemote <- function(call, frame) {
+
+  # check for active request / response
+  request  <- Sys.getenv("RSTUDIOAPI_IPC_REQUESTS_FILE", unset = NA)
+  response <- Sys.getenv("RSTUDIOAPI_IPC_RESPONSE_FILE", unset = NA)
+  secret   <- Sys.getenv("RSTUDIOAPI_IPC_SHARED_SECRET", unset = NA)
+  if (is.na(request) || is.na(response) || is.na(secret))
+    stop("internal error: callFunRemote() called without remote connection")
+
+  attr(call, "srcref") <- NULL
+
+  # ensure rstudioapi functions get appropriate prefix
+  if (is.name(call[[1L]]))
+    call[[1L]] <- call("::", as.name("rstudioapi"), call[[1L]])
+
+  # ensure arguments are evaluated before sending request
+  for (i in seq_along(call)[-1L])
+    call[[i]] <- eval(call[[i]], envir = frame)
+
+  # write to tempfile and rename, to ensure atomicity
+  data <- list(secret = secret, call = call)
+  tmp <- tempfile(tmpdir = dirname(request))
+  saveRDS(data, file = tmp)
+  file.rename(tmp, request)
```

Comment:
This shouldn't ever happen (at least in the current architecture) since RStudio won't allow the child R process to proceed from the polling loop below until that file has been removed:

https://github.com/rstudio/rstudio/pull/5391/files#diff-6fb86e81578c5a08f06852af0b868400R254-R257